SPEC.md

```diff
@@ -85,27 +85,40 @@ The operations that CNI plugins must support are:
-  - If the runtime cached the `Result` of a previous `ADD` or `GET` response for a given container, it must delete that cached response on a successful `DEL` for that container.
+  - If the runtime cached the `Result` of a previous `ADD` response for a given container, it must delete that cached response on a successful `DEL` for that container.
 
-- `GET`: Get container network configuration
+- `CHECK`: Check container's networking is correct
   - Parameters:
     - **Container ID**, as defined for `ADD`.
     - **Network namespace path**, as defined for `ADD`.
-    - **Network configuration**, as defined for `ADD`.
+    - **Network configuration** as defined for `ADD`, which must include a `prevResult` field containing the `Result` of the immediately preceeding `ADD` for the container.
     - **Extra arguments**, as defined for `ADD`.
     - **Name of the interface inside the container**, as defined for `ADD`.
   - Result:
-    - The plugin should return the same result as an `ADD` action for the same inputs.
-    - **Interfaces list**, as defined for `ADD`
-    - **IP configuration assigned to each interface**, as defined for `ADD`
-    - **DNS information**, as defined for `ADD`
-  - This action should return the same `Result` object as an `ADD` action for the same inputs. The result should not change over the lifetime of the container.
-  - The plugin should return an error if any general internal state is unexpected. For example, if the plugin's data storage is missing or corrupt, or its control plane is unavailable, it should return an error.
-  - The plugin should NOT return an error if its expected sandbox state (eg interfaces, IP addresses, routes, etc) is not found, as subsequent elements in the plugin's chain may alter sandbox state.
-  - A runtime may call `GET` at any time; but if `GET` is called for a container before an `ADD` or after a `DEL` for that container, the plugin should return error 3 to indicate the container is unknown (see [Well-known Error Codes](#well-known-error-codes) section).
-  - If the previous action for the container was `ADD` or `GET`, the runtime must add a `prevResult` field to the configuration JSON of the plugin (or all plugins in the chain), which must be the `Result` of that previous `ADD` or `GET` action in JSON format ([see below](#network-configuration-list-runtime-examples)).
+    - The plugin must return either nothing or an error.
+  - The plugin must consult the `prevResult` to determine the expected interfaces and addresses.
+  - The plugin must allow for a later chained plugin to have modified networking resources, e.g. routes.
+  - The plugin should return an error if an interface, address or route:
+    - was created by the plugin, and
+    - is listed in `prevResult`, and
+    - does not exist, or is in an invalid state.
+  - The plugin should return an error if other expected resources such as the following do not exist or are in an invalid state:
```

Comment:
I wonder if we could give a name to these resources to more explicitly call out that they're not tracked in `prevResult`.  Maybe `untracked resources`?